jujugui/static/gui/src/app/components/entity-header/entity-header.js

```diff
@@ -39,20 +45,82 @@ YUI.add('entity-header', function() {
       @returns {String} The intial state.
     */
     getInitialState: function() {
-      return {headerHeight: 0};
+      return {
+        headerHeight: 0,
+        modelList: []
+      };
+    },
+
+    componentWillMount: function() {
+      this._fetchModels();
     },
 
     componentDidMount: function() {
       this.setState({headerHeight: this.refs.headerWrapper.clientHeight});
     },
 
+    componentWillUnmount: function() {
+      this.xhrs.forEach((xhr) => {
+        xhr && xhr.abort && xhr.abort();
+      });
+    },
+
+    componentWillReceiveProps: function(nextProps) {
+      // If the user has changed then update the data.
+      if (nextProps.user.user !== this.props.user.user) {
+        this._fetchModels();
```

Comment:
Would this cause problems if `nextProps.user.user` is null (or even `nextProp.user`) due to a logout? That is, are we going to get errors in the console because we're making a request (fetchModels) that needs credentials?